marathon_acme/service.py

```diff
@@ -103,6 +105,15 @@ def log_failure(failure):
         }).addCallbacks(on_finished, log_failure, callbackArgs=[reconnects])
 
     def _sync_on_event_stream_attached(self, event):
+        if self._attached:
+            self.log.debug(
+                'event_stream_attached event received (timestamp: '
+                '"{timestamp}", remoteAddress: "{remoteAddress}"), but '
+                'already attached', timestamp=event['timestamp'],
+                remoteAddress=event['remoteAddress'])
+            return
+
+        self._attached = True
```

Comment:
I'd just clear the flag in `listen_events()`. We may end up doing an extra sync or so, but that's not really a problem.